### PR TITLE
[Decoder] Add decoder custom operation

### PR DESCRIPTION
**Pull request by gichan-jang**
It is similar to cusom-easy of the tensor filter.
Get an input tensors and make output media stream.
It can be used to create cusom media stream from tensors such as flexbuffer.
Output capability is application/octet-stream.
    
Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_decoder/tensordec-flexbuf.cc [hunk 1]
```diff
@@ -33,6 +33,26 @@
  *                          Blob | <tensor data>
  *                         }
  * }
+ *
+ * If you want to convert tensors to your own binary format of the flexbuffers,
+ * You can use custom mode of the tensor decoder.
```
Comment (zhoonit) [inline ext/nnstreamer/tensor_decoder/tensordec-flexbuf.cc:38]: Could I ask difference between custom easy and custom? 

if tensorfilter has custom-easy and custom, but custom-easy maps to custom in other types of element, I wonder wouldn't it create any confusion 

| element        | custom-hard                            | custom-easy |
|----------------|----------------------------------------|-------------------|
| tensor-filter  | custom                                 | custom-easy       |
| tensor-decoder | (is this sure if it won't be created?) | custom            |
| tensor-??? | ??? | ???           |

File: gst/nnstreamer/tensor_decoder/tensordec.c [hunk 2]
```diff
@@ -675,7 +694,7 @@ gst_tensordec_transform (GstBaseTransform * trans,
   if (G_UNLIKELY (!self->configured))
     goto unknown_format;
 
-  if (self->decoder) {
+  if (self->decoder || self->is_custom) {
```
Comment (zhoonit) [inline gst/nnstreamer/tensor_decoder/tensordec.c:697]: Code clean: do we need this if? checks at :648 makes this if always true.

File: gst/nnstreamer/tensor_decoder/tensordec.c [hunk 3]
```diff
@@ -699,9 +718,15 @@ gst_tensordec_transform (GstBaseTransform * trans,
...
       input[i].size = in_info[i].size;
     }
-
-    res = self->decoder->decode (&self->plugin_data, &self->tensor_config,
-        input, outbuf);
+    if (!self->is_custom) {
+      res = self->decoder->decode (&self->plugin_data, &self->tensor_config,
+          input, outbuf);
+    } else if (self->custom.func != NULL) {
+      res = self->custom.func (input, &self->tensor_config, outbuf);
```
Comment (zhoonit) [inline gst/nnstreamer/tensor_decoder/tensordec.c:725]: do we have to pass custom.data here?

File: gst/nnstreamer/tensor_decoder/tensordec.c [hunk 4]
```diff
@@ -893,13 +932,57 @@ gst_tensordec_transform_size (GstBaseTransform * trans,
...
+/**
+ * @brief Unregisters a callback for tensor_decoder custom condition
+ * @return 0 if success. -ERRNO if error.
+ */
+int
+nnstreamer_decoder_custom_unregister (const gchar * name)
+{
+  decoder_custom_cb_s *ptr;
+
+  ptr = (decoder_custom_cb_s *) get_subplugin (NNS_CUSTOM_DECODER, name);
```
Comment (zhoonit) [inline gst/nnstreamer/tensor_decoder/tensordec.c:980]: Is it okay if we don't do null check on this ptr?

File: gst/nnstreamer/tensor_decoder/tensordec.c (hunk 2 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_decoder/tensordec.c:697]: Well.. The two conditions are not directly related.

File: gst/nnstreamer/tensor_decoder/tensordec.c (hunk 4 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_decoder/tensordec.c:981]: I think it would be unnecessary.

File: gst/nnstreamer/tensor_decoder/tensordec.c (hunk 3 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_decoder/tensordec.c]: Thanks. Updated.

File: ext/nnstreamer/tensor_decoder/tensordec-flexbuf.cc (hunk 1 above)
Comment (zhoonit) [reply ext/nnstreamer/tensor_decoder/tensordec-flexbuf.cc:38]: I see, i guess this can be handled in different PR.

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: gst/nnstreamer/tensor_decoder/tensordec.c [hunk 5]
```diff
@@ -555,7 +570,9 @@ gst_tensordec_get_property (GObject * object, guint prop_id,
       g_value_set_boolean (value, self->silent);
       break;
     case PROP_MODE:
-      if (self->decoder)
+      if (self->is_custom)
+        g_value_set_string (value, "custom");
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_decoder/tensordec.c:574]: consider 'custom-code' now.

Comment (gichan-jang) [reply gst/nnstreamer/tensor_decoder/tensordec.c]: right. updated.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**